tests/test_data/test_submission_portal_translator.py

```diff
@@ -234,18 +240,37 @@ def test_get_from():
     assert translator._get_from(metadata, ["one", "some_empty"]) == ["one", "three"]
 
 
-def test_get_dataset(test_minter):
-    random.seed(0)
+def test_get_dataset(test_minter, monkeypatch):
+    # OmicsProcess objects have an add_date and a mod_date slot that are populated with the
+    # current date. In order to compare with a static expected output we need to patch
+    # the datetime.now() call to return a predefined date.
+    the_time = datetime.datetime(2023, 10, 17, 9, 0, 0)
+
+    class FrozenDatetime(datetime.datetime):
+        @classmethod
+        def now(cls, **kwargs):
+            return the_time
+
+    monkeypatch.setattr(
+        "nmdc_runtime.site.translation.submission_portal_translator.datetime",
+        FrozenDatetime,
+    )
+
+    mongo_db = get_mongo_test_db()
     with open(
         Path(__file__).parent / "test_submission_portal_translator_data.yaml"
     ) as f:
         test_datasets = yaml.safe_load_all(f)
 
         for test_data in test_datasets:
+            random.seed(0)
```

Comment:
I'm not seeing where randomness comes into play in this test. Maybe one of the called functions relies on it.

Reply:
The `test_minter` [fixture](https://github.com/microbiomedata/nmdc-runtime/blob/29c23ffb54ca2ce1da23bb43d15e658478ceee44/tests/test_data/conftest.py#L10) generates random IDs. Resetting the seed keeps the generated IDs stable between test runs.

Reply:
Ah, OK. That makes sense. Thanks!